### DISCUSSION TITLE
Wrangling sebelum pembuatan scatterplot

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#566] [Q&A] [answered]
Sebelumnya, terima kasih kepada tim Algoritma atas bantuan mengenai cara menggabungkan banyak data frame menjadi 1.  Pada saat ini saya kebingungan mengenai cara wranglingnya. Terlampir berikut adalah bentuk data.frame yang saya miliki:
![image](https://user-images.githubusercontent.com/122977451/220849084-a9978eda-bf9d-4327-a092-e16cb1548595.png)

Rencana saya adalah untuk melihat korelasi antara daily return (DR) dari saham-saham (Ticker) yang berbeda. Dari yang saya pahami, kalau membuat scatterplot dengan ggplot adalah untuk membuat "kanvas" antara 2 kolom berbeda yang kemudian dijadikan sumbu x dan y. 

Nah masalah dengan data frame saya adalah saya ingin menjalankan korelasi dengan data-data yang berada pada 1 kolom yang sama (DR), yang kemudian dibedakan dengan Ticker. Sejauh ini saya baru kepikiran bentuk wranglingnya seperti ini, dan hasilnya pun sepertinya janggal:
![image](https://user-images.githubusercontent.com/122977451/220849978-2d113c73-3eac-46a6-9073-0cac3740a0f4.png)

Apakah saya perlu mengubah data frame tersebut menjadi memanjang ke kanan, yang dimana kolom dibedakan berdasarkan saham, atau apakah ada langkah lain yang perlu dilakukan dalam proses wrangling saya?

Terima kasih lagi sebelumnya

**Comment by fafilia** [comment, marked as answer]
Dear @AGelfP ,

Untuk membuat scatter plot sesuai yang Bapak inginkan, silahkan untuk melakukan reshaping dataframe `for_plot` ke format wide menggunakan fungsi `pivot_wider()` setelah itu Bapak dapat membuat langsung scatter plotnya dari kolom-kolom Ticker yang telah di reshape.
